src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/EditAndContinue/EditAndContinueProvider.cs

```diff
@@ -7,78 +7,217 @@
 using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
 using Microsoft.VisualStudio.ProjectSystem.VS;
 using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
 
-using EncInterop = Microsoft.VisualStudio.LanguageServices.Implementation.EditAndContinue.Interop;
+// We use the version of IVsENCRebuildableProjectCfg2/IVsENCRebuildableProjectCfg4 from Roslyn because the one in the SDK is defined wrong
+using IVsENCRebuildableProjectCfg2 = Microsoft.VisualStudio.LanguageServices.Implementation.EditAndContinue.Interop.IVsENCRebuildableProjectCfg2;
+using IVsENCRebuildableProjectCfg4 = Microsoft.VisualStudio.LanguageServices.Implementation.EditAndContinue.Interop.IVsENCRebuildableProjectCfg4;
+using ENC_BREAKSTATE_REASON = Microsoft.VisualStudio.LanguageServices.Implementation.EditAndContinue.Interop.ENC_BREAKSTATE_REASON;
 
 namespace Microsoft.VisualStudio.ProjectSystem.Managed.VS.EditAndContinue
 {
-
-    [ExportProjectNodeComService(typeof(IVsENCRebuildableProjectCfg))]
+    [ExportProjectNodeComService(typeof(IVsENCRebuildableProjectCfg), typeof(IVsENCRebuildableProjectCfg2), typeof(IVsENCRebuildableProjectCfg4))]
     [AppliesTo(ProjectCapability.EditAndContinue)]
-    internal class EditAndContinueProvider : IVsENCRebuildableProjectCfg
+    internal class EditAndContinueProvider : IVsENCRebuildableProjectCfg, IVsENCRebuildableProjectCfg2, IVsENCRebuildableProjectCfg4, IDisposable
```

Comment:
:bulb: Mark this `sealed` rather than suppress CA1063.

Reply:
CA1063 is designed for public types and should not be firing on internal types.

Reply:
Hence the bug.